src/Features/LanguageServer/ProtocolUnitTests/Ordering/RequestOrderingTests.cs

```diff
@@ -120,6 +121,32 @@ public async Task ThrowingTaskDoesntBringDownQueue()
             Assert.All(responses, r => Assert.True(r.EndTime > r.StartTime));
         }
 
+        [Fact]
+        public async Task LongRunningSynchronousNonMutatingTaskDoesNotBlockQueue()
+        {
+            var requests = new[] {
+                new TestRequest(LongRunningNonMutatingRequestHandler.MethodName),
+                new TestRequest(MutatingRequestHandler.MethodName),
```

Comment:
hm- should the mutating request be included?  I guess it doesn't actually matter as previously even the mutating request would be blocked by a long running synchronous non-mutating

Reply:
I figured I'd write one test to encompass interactions with both mutating/non-mutating since prior to this it'd block both 😄